### PR TITLE
Add documentation for 'User Guide' with available options

### LINKED ISSUE #715: Convert global or limit-related variables into configurations.
https://github.com/databricks/koalas/commit/94adf06d622aaac1eb05c03407db8a1abc01632c added configuration base like Pandas. See https://pandas.pydata.org/pandas-docs/stable/user_guide/options.html

Currently we have multiple variables that should be converted into configuration values. We should identify and convert it to configuration _all_.

- [x] Support typed values in configurations (#714)
- [x] Add 'display.max_rows' (#714)
- [x] Add 'compute.shortcut_limit' (#717) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/frame.py#L373-L379
- [x] Add 'compute.max_rows'  (#721) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/frame.py#L1463-L1470 (I think we should remove `limit` parameter).
- [x] Add 'compute.shortcut_limit' (#717) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/groupby.py#L806-L812
- [x] Add 'compute.shortcut_limit' (#717) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/groupby.py#L1512-L1518
- [x] Add 'plotting.max_rows'  (#728) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/plot.py#L54
- [x] Add 'plotting.sample_ratio' (#737) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/plot.py#L85
- [x] Add 'compute.ops_on_diff_frames' (#725) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/utils.py#L63
- [x] Add 'compute.default_index_type' (#723) https://github.com/databricks/koalas/blob/8360c586f5ad4c8847fe394eefbd77b48d95f687/databricks/koalas/internal.py#L477
- [x] Add 'compute.max_rows' (#736) https://github.com/databricks/koalas/pull/712/files#diff-5ba644c40e914732959b9b4867f30c53R1644
- [x] Add 'compute.shortcut_limit' (#739) https://github.com/databricks/koalas/pull/732/files#diff-5ba644c40e914732959b9b4867f30c53R1605
- [ ] Add documentation (#743). See https://pandas.pydata.org/pandas-docs/stable/user_guide/options.html

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds 'User Guide' with available options.
Can be verified as below:

```bash
./dev/lint-python
open docs/build/html/index.html
```

Resolves #715

### CODE REVIEW

File: docs/source/user_guide/options.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+There are several types of the default index that can be configured by `compute.default_index_type` as below:
+
+**sequence**: It implements a sequence that increases one by one, by Window function without
+specifying partition. Therefore, it can end up with whole partition in single node.
+This index type should be avoided when the data is large. This is default. See example below:
+
+.. code-block:: python
+
+    >>> ks.range(3).index  # doctest: +SKIP
```
Comment (HyukjinKwon) [inline docs/source/user_guide/options.rst]: Let me remove this skip flag and clean up a bit more tomorrow.

File: docs/source/user_guide/options.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+Koalas has an options system that lets you customize some aspects of its behaviour,
+display-related options being those the user is most likely to adjust.
+
+Options have a full "dotted-style", case-insensitive name (e.g. ``display.max_rows``).
+
+The API is composed of 3 relevant functions, available directly from the ``koalas``
+namespace:
+
+* :func:`~koalas.get_option` / :func:`~koalas.set_option` - get/set the value of a single option.
+* :func:`~koalas.reset_option` - reset one or more options to their default value.
```
Comment (ueshin) [inline docs/source/user_guide/options.rst]: These functions are not linked. Seems like we can link with something like the following:

```diff
diff --git a/docs/source/user_guide/options.rst b/docs/source/user_guide/options.rst
index 783ca31..503c89d 100644
--- a/docs/source/user_guide/options.rst
+++ b/docs/source/user_guide/options.rst
@@ -1,6 +1,7 @@
 ====================
 Options and settings
 ====================
+.. currentmodule:: databricks.koalas

 Overview
 --------
@@ -12,8 +13,8 @@ Options have a full "dotted-style", case-insensitive name (e.g. ``display.max_ro
 The API is composed of 3 relevant functions, available directly from the ``koalas``
 namespace:

-* :func:`~koalas.get_option` / :func:`~koalas.set_option` - get/set the value of a single option.
-* :func:`~koalas.reset_option` - reset one or more options to their default value.
+* :func:`get_option` / :func:`set_option` - get/set the value of a single option.
+* :func:`reset_option` - reset one or more options to their default value.
```

File: docs/source/user_guide/options.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+This can be enabled by setting `compute.ops_on_diff_frames` to `True` to allow such cases.
+See the examples below.
+
+.. code-block:: python
+
+    >>> import databricks.koalas as ks
+    >>> kdf = ks.range(5)
+    >>> kser_a = ks.Series([1, 2, 3, 4])
+    >>> # 'kser_a' is not from 'kdf' DataFrame. So it is considered as a Series not from 'kdf'.
+    >>> kdf['new_col'] = kser_a
```
Comment (ueshin) [inline docs/source/user_guide/options.rst]: We should show enabling the option to `True`?

File: docs/source/user_guide/options.rst (hunk 2 above)
Comment (HyukjinKwon) [reply docs/source/user_guide/options.rst]: nice!

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**